### PR TITLE
fix(to-require-context): fixed regex to glob conversion

### LINKED ISSUE #10178: Bug in glob to require-context regex: storybook 5.3 attempts to load stories from node_modules
**Describe the bug**
i upgraded my monorepo to 5.3.17 (from 5.2.8) and SB now tries to load stories from storybook's own node_modules directories.

Im hoping its just me doing something wrong but from what i can see this is an issue with SB and a breaking change from pre 5.3.

my .storybook/config.js had this configure:

```
configure(require.context("../packages",
	true,
	/\.stor(y|ies)\.js/), module);
```
which i translated to main.js:

```
module.exports = {
    stories: [
        "../packages/**/*.stories.js"
    ]
...
}
```

when i run sb i get this error:

```
ReferenceError: exports is not defined
    at Module../packages/ui/upload-button/node_modules/@storybook/components/dist/ActionBar/ActionBar.stories.js (http://localhost:9111/vendors~main.8d3b8dd16022b2b66155.bundle.js:128649:23)
    at __webpack_require__ 
```

from debugging i can see the resulting configuration passed to the require context is: 

> ./packages sync recursive ^\.\/(?:(?:(?!\.)(?:(?:(?!(?:|\/)\.).)*?)\/)?(?!\.)(?=.)[^\/]*?\.stories\.js\/?)$

my repo has this structure:

root
 --- packages
 ----- pkg-a
 ----- pkg-b
 ----- ui
 ------- ui-pkg-a
 ------- ui-pkg-b

the UI packages have the story file in their root:

Screen Shot 2020-03-21 at 11 24 53

but since each package has its own node_modules folder, SB attempts to load stories found in it (through symlink). it finds the stories from SB's own code.

When I tried to improve the glob pattern to be more precise. for example:

> "../packages/ui/\*/\*.stories.js"

SB doesnt find any stories...

when i run this pattern using glob it correctly returns only the relevant files:

```
console.log(glob.sync("./packages/ui/*/*.stories.js"));
```

returns:

[ './packages/ui/chunked-uploady/ChunkedUploady.stories.js',
  './packages/ui/upload-button/UploadButton.stories.js',
  './packages/ui/upload-drop-zone/UploadDropZone.stories.js',
  './packages/ui/upload-preview/UploadPreview.stories.js',
  './packages/ui/upload-url-input/UploadUrlInput.stories.js' ]

when i tried with exclude:

> "../packages/**/!(node_modules/)*.stor(y|ies).js"

SB still loads all the files.

**To Reproduce**

code is available on my branch.
https://github.com/yoavniran/react-uploady/blob/storybook-upgrade/.storybook/main.js

1. check out repo on branch: storybook-upgrade
2. run: yarn
3. run: yarn ssb
4. See error in browser

**Expected behavior**
SB shouldnt load stories from node_modules

there is also 0 documentation around the stories path configuration which would be very useful. 

**System:**

 System:
    OS: macOS 10.15.3
    CPU: (12) x64 Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
  Binaries:
    Node: 10.15.2 - ~/.nvm/versions/node/v10.15.2/bin/node
    Yarn: 1.19.1 - ~/.yarn/bin/yarn
    npm: 6.13.4 - ~/.nvm/versions/node/v10.15.2/bin/npm
  Browsers:
    Firefox: 73.0.1
    Safari: 13.0.5
  npmPackages:
    @storybook/addon-actions: ^5.3.17 => 5.3.17 
    @storybook/addon-console: ^1.2.1 => 1.2.1 
    @storybook/addon-knobs: ^5.3.17 => 5.3.17 
    @storybook/addon-storysource: ^5.3.17 => 5.3.17 
    @storybook/react: ^5.3.17 => 5.3.17

### PR DESCRIPTION
**Pull request by valerybugakov**
Issue: described here: https://github.com/storybookjs/storybook/issues/10322

## What I did

Fixed conversion from glob to regex. Added tests for `to-require-context` module.

## How to test

```sh
npx jest lib/core/src/server/preview/to-require-context.test.js
```

### CODE REVIEW

**Comment by valerybugakov** [comment]
`svelte-kitchen-sink` tests are failing in CircleCI. Let me know what should be done to fix them

**Comment by ndelangen** [comment]
There's something odd happening here:
https://circleci.com/gh/storybookjs/storybook/207662

This glob seems to be broken by this change:
https://github.com/storybookjs/storybook/blob/4a3d16103656a78b9fb377107bb755d2de80a7fb/examples/cra-kitchen-sink/.storybook/main.js#L18

File: lib/core/src/server/preview/to-require-context.test.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    ];
+
+    const { match } = toRequireContext(inputGlob);
+    const regex = new RegExp(match.slice(1, -1));
+
+    const isMatchedForCorrectPaths = correctPaths.every((path) => regex.test(path));
+    const isMatchedForIncorrectPaths = incorrectPaths.every((path) => regex.test(path));
+
+    expect(isMatchedForCorrectPaths).toBe(true);
+    expect(isMatchedForIncorrectPaths).toBe(false);
```
Comment (shilman) [inline lib/core/src/server/preview/to-require-context.test.js:29]:
```
> [false, true].every(x => x)
false
```

I think you want:

```js
const isMatchedForIncorrectPaths = incorrectPaths.every((path) => !regex.test(path));
...
expect(isMatchedForIncorrectPaths).toBe(true);
```
no?

**Review by shilman — CHANGES_REQUESTED** [review]
See suggestion

File: lib/core/src/server/preview/to-require-context.test.js (hunk 1 above)
Comment (valerybugakov) [reply lib/core/src/server/preview/to-require-context.test.js:29]: right, thanks, that was the intention 👍

**Comment by valerybugakov** [comment]
This [issue](https://github.com/micromatch/micromatch/issues/110) was [fixed](https://github.com/micromatch/nanomatch/commit/770f54c59b3a4b60a92dd1c2a396018b5596f13a) in `nanomatch`. It works correctly without any replacements using `makeRe` from `nanomatch`.

**Review by ndelangen — APPROVED** [review]

**Pull request merged**